src/CoherenceBuild/CoherenceVerifier.cs

```diff
@@ -1,19 +1,43 @@
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Runtime.Versioning;
 
 namespace CoherenceBuild
 {
     public class CoherenceVerifier
     {
-        public static bool VerifyAll(ProcessResult result)
+        private readonly List<PackageInfo> _packages;
+        private readonly Dictionary<string, PackageInfo> _packageLookup;
+        private readonly bool _verifyPackages;
+        private readonly bool _verifyPartnerPackages;
+
+        public CoherenceVerifier(
+            List<PackageInfo> packages,
+            bool verifyPackages,
+            bool verifyPartnerPackages)
         {
-            foreach (var productPackageInfo in result.ProductPackages.Values)
+            _packages = packages;
+            _packageLookup = _packages.ToDictionary(p => p.Identity.Id, StringComparer.OrdinalIgnoreCase);
+            _verifyPackages = verifyPackages;
+            _verifyPartnerPackages = verifyPackages && verifyPartnerPackages;
```

Comment:
This is confusing, if you don't expect to validate only partner packages then write use an enum instead of two bools